Implement fetch_bundle for TDR plugin (#1844)

noah-aviel-dove opened the pull request.

noah-aviel-dove commented:
Rebasing on #1857 to address review feedback that affects shared commits. This may somewhat invalidate peer review.

noah-aviel-dove commented:
Reminder to self: fix manifest `size` for data files before requesting Hannes review.

noah-aviel-dove commented:
Re-requesting peer review due to substantial changes following rebase and addition of file descriptors.

achave11 commented on src/azul/bigquery.py:
```diff
@@ -28,6 +29,10 @@ def assert_table_exists(self, dataset_name: str, table_name: str) -> None:
         raise NotImplementedError
 
+    @abstractmethod
+    def create_table(self, dataset_name: str, table_name: str, schema: JSONs, rows: JSONs) -> None:
```
Is this missing  docstrings as per the other two methods in this class?

noah-aviel-dove commented:
> Regarding the changes in general, it seems that there is string usage in both single quotes and double quotes, did you apply any specific JSON/str/dict standards or was it an overlook?

I _tried_ to apply a specific standard but overthought it and kept changing my delimiters until I gave up on consistency. Simplified and fixed now.
